File: week-1/two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable = {}
        
        for i in range(len(nums)):
            if target - nums[i] in hashtable:
                return [i, hashtable[target-nums[i]]]
            hashtable[nums[i]] = i
            
        return []

File: week-1/two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hashtable = {};
    let result = [];
    nums.every((num, i) => {
        if (hashtable.hasOwnProperty(target-num)) {
            result = [i, hashtable[target-num]];
            return false;
        }
        hashtable[num] = i;
        return true;
    });
    return result;
};
